clang/tools/driver/cc1depscan_main.cpp

```diff
@@ -1096,7 +1096,9 @@ int cc1depscand_main(ArrayRef<const char *> Argv, const char *Argv0,
     ShutDown.store(true);
   }
 
-  return 0;
+  // Exit instead of return. Otherwise, it will wait on ~ThreadPool() which will
+  // never return since all threads might still be sleeping on ::accept().
+  ::exit(0);
```

Comment:
Are we sure it's safe to `exit` here if there are threads that haven't sync'd? We're going to run global destructors, etc. that could be accessed by the thread.

Reply:
To get out of the loop above, you need to hit `NumRunning` is 0, which means all threads are sitting on `accept()`. Then `ShutDown.store(true)` should prevent threads spin up a new job.

So technically, yes, thread should either wait on accept or just returned. Will it crash in all edge cases? I don't know. We need concurrency to save us.